master/buildbot/steps/source/p4.py

```diff
@@ -108,10 +108,6 @@ def __init__(self, mode='incremental',
             config.error('p4branch should not end with a trailing / [p4branch = {}]'.format(
                     p4branch))
 
-        if (p4branch or p4extra_views) and not p4base:
```

Comment:
It's not obvious to me that this is dead code. It seems that this case is not handled by the other conditionals around, is it?

Reply:
The logic is a bit obscure.
`not p4base` requires `p4viewspec`:
https://github.com/buildbot/buildbot/blob/913950c1f996c37e4aff1707e261af6a63b59db4/master/buildbot/steps/source/p4.py#L90
and `p4viewspec` requires `not (p4branch or p4extra_views)`:
https://github.com/buildbot/buildbot/blob/913950c1f996c37e4aff1707e261af6a63b59db4/master/buildbot/steps/source/p4.py#L93
I value code coverage more than having unreachable safety checks

Reply:
Indeed, thanks for explanation.